### ISSUE TITLE
Create new type of accounts

### ISSUE THREAD

**Issue by dmmleal** [moneymanagerex/moneymanagerex#2017]
Hello, many thanks for providing Money Manager EX.
Is there any chance to allow the user to create their own types of bank accounts, and not only be restricted to those that already come predefined?
In this way we could know the total value of deposits per bank, considering each type account as a banking institution.
For example:
Account type 1 (Bank A)
Checking account - €20,00
     1 year deposit - €120,00
                    Total - €140,00
Account type 2 (Bank B)
Checking account - €40,00
6 months deposit - €200,00
                    Total - €240,00
Instead of having: “Bank Accounts”, “Term Accounts”, “Loan Accounts”, “Cash Accounts”, that can not be edited.
Thanks a lot.

**Comment by vomikan** [comment]
I don't understand your explanation.
You may use only one type of accounts for all general purposes except 'Investment'.

**Comment by slodki** [comment]
Because #1866

**Comment by dmmleal** [comment]
Because I have different accounts in different banks, and I would like to divide them by banks and not by account.
If I was able to create my own accounts, it would enable me to name them by bank name and the Money manager ex would, therefore, sum the amount of money that I have in each bank.

**Comment by vomikan** [comment]
This task quite easy to solve by custom report. Singing like that:
https://github.com/moneymanagerex/general-reports/issues/42

**Comment by dmmleal** [comment]
Thank you very much.
It was very simple after all.
SQL
select a.HELDAT
, a.INITIALBAL + total(t.TRANSAMOUNT) as 'TOTAL'
, c.baseconvrate
, (a.INITIALBAL + total(t.TRANSAMOUNT))*c.baseconvrate as BaseCurrencyBalance
    from
    (select ACCOUNTID, STATUS,
    (case when TRANSCODE='Deposit' then TRANSAMOUNT else -TRANSAMOUNT end) as TRANSAMOUNT
    from CHECKINGACCOUNT_V1
    union all    
select TOACCOUNTID, STATUS, TOTRANSAMOUNT
    from CHECKINGACCOUNT_V1
    where TRANSCODE='Transfer') as t
    inner join ACCOUNTLIST_V1 as a on a.ACCOUNTID=t.ACCOUNTID
    inner join CURRENCYFORMATS_V1 as c on a.CURRENCYID=c.CURRENCYID
    where a.Status='Open'
and t.STATUS<>'V'
and a.accountname like '%'
group by a.HELDAT
order by HELDAT

LUA
local total = 0;

function handle_record(record)
    total = total + record:get("BaseCurrencyBalance");;
end

function complete(result)
    result:set("Total", total);
end

TEMPLATE
<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <meta http - equiv = "Content-Type" content = "text/html">
    <title><TMPL_VAR REPORTNAME></title>
    <script src = "sorttable.js"></script>
    <link href = "master.css" rel = "stylesheet">
    
</head>
<body>
    <div class="container">
        <h3>Poupança por Banco</h3>
        <p><TMPL_VAR TODAY></p>
        <div class="row">
            <div class="col-xs-2"></div>
             <div class="col-xs-4">
                <table class="table sortable">
                <thead>
                  <tr>
                  <th class="i18n">Banco</th>
                    <th class="i18n">Balanço</th>
                    <th class="i18n">Racio</th>
                    <th class="i18n">Total</th>
                  </tr>
                </thead>
                <tbody>
                    <TMPL_LOOP NAME=CONTENTS>
                    <tr>
                        <th><TMPL_VAR HELDAT></th>
                        <td class="money" sorttable_customkey='<TMPL_VAR Balance>'><TMPL_VAR Balance></td>
                        <td class="money" sorttable_customkey='<TMPL_VAR baseconvrate>'><TMPL_VAR baseconvrate></td>
                        <td class="money" sorttable_customkey='<TMPL_VAR BaseCurrencyBalance>'><TMPL_VAR BaseCurrencyBalance></td>
                    </tr>
                    </TMPL_LOOP>
                </tbody>
                <tfoot>
                    <tr class="total">
                        <td class="i18n">Total</td>
                        <td></td>
                        <td></td>
                        <td class="money"><TMPL_VAR Total></td>
                    </tr>
                </tfoot>
                </table>
            </div>
            <table class="table">
                <tbody>
                <TMPL_LOOP ERRORS>
                    <tr><td><h3><span class="label label-danger">Error</span></h3></td></tr>
                    <tr><td><TMPL_VAR ERROR></td></tr>
                </TMPL_LOOP>
                </tbody>
            </table>
        </div>
    </div>   
</body>
<script>
    
    function currency(n) {
        n = parseFloat(n);
        n =  isNaN(n) ? 0 : n.toFixed(2);
        var out = n.toString().replace(".", "|");
        out = out.toString().replace(/\B(?=(\d{3})+(?!\d))/g, "<TMPL_VAR GROUP_SEPARATOR>");
        out = out.replace("|", "<TMPL_VAR DECIMAL_POINT>");
        return out;
    }
    var elements= document.getElementsByClassName("money");
    for (var i = 0; i < elements.length; i++) {
        var element = elements[i];
        element.style.textAlign='right';
        element.style.whiteSpace = 'nowrap';
        if (element.innerHTML.indexOf("-") > -20) {
            element.style.color="#CC0000";
        } 
        element.innerHTML = currency(element.innerHTML);
    }
</script>
</html>
